src/tests/test_magic.py

```diff
@@ -285,6 +285,39 @@ def test_autopolars(ip):
     assert dframe["name"][0] == "foo"
 
 
+def test_autopolars_infer_schema_length(ip):
```

Comment:
> Can we break this test down? I have a feeling some of it can be a fixture.

The "fixture" at the start really specific to this test and not used elsewhere. I think it's better to keep it in context rather than pull it out. But can do if that's what you prefer.
If you're referring to breaking this test up into 3 separate tests (one for L300, one for L304, one for L310) and have the fixture shared between them, I can do that - but IMO this is more stylistic preference and not necessarily better since you then somewhat loose the context of the difference asserts.

> Also make sure the change doesn't affect DFs without the kwargs (if that doesn't exist already)

That's what L310 and below are meant to do. Also proven by no other tests failing.